pkg/generate/app/sourcelookup.go

```diff
@@ -15,6 +16,11 @@ import (
 	s2igit "github.com/openshift/source-to-image/pkg/scm/git"
 )
 
+var (
+	argumentGit         = regexp.MustCompile("^(http://|https://|git@|git://).*(?:#([a-zA-Z0-9]*))?$")
+	argumentGitProtocol = regexp.MustCompile("^(git@|git://)")
```

Comment:
@gabemontero I'm reintroducing these to fix a regression, perhaps you have in mind a better solution?
Should we also fix something in S2I?